sessionrecording/connect.go

```diff
@@ -130,8 +130,23 @@ func SessionRecordingClientForDialer(dialCtx context.Context, dial func(context.
 				cancel()
 			}
 		}()
-		return dial(perAttemptCtx, network, addr)
+
+		conn, err := dial(perAttemptCtx, network, addr)
```

Comment:
let's make this a separate PR, once we prove that these keepalives actually work